src/app/reporting/Engine.cpp

```diff
@@ -903,6 +903,11 @@ CHIP_ERROR Engine::ScheduleBufferPressureEventDelivery(uint32_t aBytesWritten)
 
 CHIP_ERROR Engine::ScheduleEventDelivery(ConcreteEventPath & aPath, uint32_t aBytesWritten)
 {
+    if (InteractionModelEngine::GetInstance()->mEventPathPool.Allocated() == 0)
```

Comment:
So this is trying to optimize out scheduling things if we literally have no read handlers right now that care about any events?  The idea being that evicting events that no one cares about is fine?  And we can't check read handlers against aPath because the things we will be evicting have some other path(s)?

Might be worth documenting that here.

Reply:
If we literally have no read handlers right now that care about any events, we don't need to call schedule run for event. If schedule run is called, actually we would not delivery events as well. Just wanna save one schedule run here.